### PR TITLE
perf(maintenance): ⚡️ Improve code logic.

### PR DESCRIPTION
**Pull request by ss-o**
## Report for z-shell/zsh-zoxide plugin

```zsh
------------------------------------
Source zsh-zoxide.plugin.zsh (reporting enabled)
Autoload .zi-prepare-zoxide with options -U -z
Autoload .zsh-prepare-zoxide with options -U -z
Saving `compdef -d x' for replay
Saving `compdef -d xi' for replay
Saving `compdef __zoxide_z_complete x' for replay
```

### ―― Functions created:

```zsh
x xi
.zi-prepare-zoxide __zoxide_cd
__zoxide_hook __zoxide_pwd
__zoxide_z __zoxide_z_complete
__zoxide_z_complete_helper __zoxide_zi
.zsh-prepare-zoxide 
```

### ―― Variables added or redefined:

```zsh
exit_code [ "" -> scalar ]
_ZO_CMD_PREFIX [ "" -> scalar ]
chpwd_functions [ "" -> array ]
__zoxide_z_prefix [ "" -> scalar ]
_ZO_DATA_DIR [ "" -> scalar-export ]
```

### ―― Completions:

```zsh
_zoxide [enabled]
_zoxide [enabled]
_cmd.rs [enabled]
```

### CODE REVIEW

**Comment by ss-o** [comment]
@0xMRTT, what do you think, boss? Does this fit with your style guide?

File: zsh-zoxide.plugin.zsh [hunk 1]
```diff
@@ -15,23 +17,46 @@ if [[ $PMSPEC != *f* ]]; then
...
 fi
 
+# https://wiki.zshell.dev/community/zsh_plugin_standard#global-parameter-with-capabilities
+if [[ $PMSPEC == *P* ]]; then
+  _ZO_DATA_DIR=${ZPFX}/share
+fi
+
 # Autoload functions
-autoload -Uz "${0:h}/functions"/.*(.:t)
+autoload -Uz "${Plugins[ZSH_ZOXIDE]}"/functions/.*(.:t)
```
[the remark this answers is not in the document]
Comment (ss-o) [reply zsh-zoxide.plugin.zsh:26]: Merci beaucoup monsieur :octocat:

File: functions/.zi-prepare-zoxide [hunk 2]
```diff
@@ -1,27 +1,37 @@
...
-# Set zoxide commands x, xi when using with Zi.
-if (( ${ZI[SOURCED]} )) && \
-(( ${+functions[zi]} )) && [[ -z ${_ZO_CMD_PREFIX} ]]; then
-  typeset -g _ZO_CMD_PREFIX=x
+# Manpages are copied to set / default location.
+if [[ -d ${ZI[MAN_DIR]} ]] && [[ -d ${Plugins[ZSH_ZOXIDE]:?}/man/man1 ]]; then
+  # Copy manpages if existing, else show error
+  command cp "${Plugins[ZSH_ZOXIDE]}"/man/man1/* "${ZI[MAN_DIR]}"/man1/ || {
+  +zi-message "{error}Failed to install manpages.{rst}"
+  ret=201
```
Comment (ss-o) [inline functions/.zi-prepare-zoxide:30]:
```suggestion
    +zi-message "{error}Failed to install manpages.{rst}"
    ret=201
```

**Comment by 0xMRTT** [comment]
> @0xMRTT, what do you think, boss? Does this fit with your style guide?

I think 👍

**Comment by ss-o** [comment]
> > @0xMRTT, what do you think, boss? Does this fit with your style guide?
> 
> I think +1

Have you tested? :eyes:

**Comment by ss-o** [comment]
OMG pressed the wrong button and closed it for a second :smile:

**Comment by 0xMRTT** [comment]
> > > @0xMRTT, what do you think, boss? Does this fit with your style guide?
> > 
> > 
> > I think +1
> 
> Have you tested? 👀

I checked if it follow the style guides. I can't test for now. Sorry

**Comment by ss-o** [comment]
> The `PMSPEC` parameter must be set for plugin managers. Is this intentional?

In my opinion, it is always better to run into an error and see it, rather than have a hidden error. Having this in mind seems reasonable for plugin managers to declare the capabilities so the plugin can be installed as expected.

It's still in a form of an idea as haven't fully figured out the implementation. :octocat:

**Comment by ss-o** [comment]
This pull request has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
To build on this closed PR, you can clone it using this method:
  - https://stackoverflow.com/a/14969986
Then open a new PR, referencing this closed PR in your message.

**Pull request merged**